### PR TITLE
Fix epoch/slot calculation after shelley parameter change

### PR DESCRIPTION
**Pull request by ppershing**

### CODE REVIEW

File: src/textUtils.c [hunk 1]
```diff
@@ -53,13 +53,35 @@ size_t str_formatAdaAmount(uint64_t amount, char* out, size_t outSize)
 	return rawSize;
 }
 
+
+// TODO: This is valid only for mainnet
+static struct {
+	uint64_t startBlockNumber;
```
Comment (janmazak) [inline src/textUtils.c:59]: Does this mean absolute slot number? There can be fewer blocks than slots.

File: src/textUtils.c [hunk 2]
```diff
@@ -53,13 +53,35 @@ size_t str_formatAdaAmount(uint64_t amount, char* out, size_t outSize)
...
+};
+
 size_t str_formatTtl(uint64_t ttl, char* out, size_t outSize)
 {
 	ASSERT(outSize < BUFFER_SIZE_PARANOIA);
 
-	const uint64_t SLOTS_IN_EPOCH = 21600;
-	uint64_t epoch = ttl / SLOTS_IN_EPOCH;
-	uint64_t slotInEpoch = ttl % SLOTS_IN_EPOCH;
+	unsigned i = 0;
```
Comment (janmazak) [inline src/textUtils.c:71]: uint64_t?

Comment (ppershing) [reply src/textUtils.c:71]: not needed. we are indexing a rather small array

File: src/textUtils.c [hunk 3]
```diff
@@ -53,13 +53,35 @@ size_t str_formatAdaAmount(uint64_t amount, char* out, size_t outSize)
...
+	while (ttl < EPOCH_SLOTS_CONFIG[i].startBlockNumber) {
+		i++;
+		ASSERT(i < ARRAY_LEN(EPOCH_SLOTS_CONFIG));
```
Comment (janmazak) [inline src/textUtils.c:74]: Is inability to display TTL enough to crash and not sign a tx?...

File: src/textUtils.c (hunk 1 above)
Comment (ppershing) [reply src/textUtils.c:59]: yep, that is absolute slot number. I did not want to confuse slots (which are usually meant per-epoch) with this absolute number

File: src/textUtils.c (hunk 3 above)
Comment (ppershing) [reply src/textUtils.c:74]: Yes. Plus, this is actually buffer overflow check

**Pull request merged**